File: gripl/gripl-backend/src/main/kotlin/de/mertendieckmann/griplbackend/model/dto/Prompt.kt
package de.mertendieckmann.griplbackend.model.dto

/**
 * Represents a prompt, which is a named collection of prompt versions.
 *
 * @property id Unique identifier
 * @property name Descriptive name of the prompt
 * @property createdAt Timestamp when the prompt was created
 * @property updatedAt Timestamp when the prompt was last modified
 */
data class Prompt(
    val id: Long,
    val name: String,
    val createdAt: String, 
    val updatedAt: String
) {
    companion object {
        fun fromRow(id: Long, name: String, createdAt: String, updatedAt: String): Prompt {
            return Prompt(id, name, createdAt, updatedAt)
        }
    }
}

File: gripl/gripl-frontend/src/models/dto/Prompt.ts
/**
 * Represents a prompt as received from the backend API.
 * A prompt is a named collection of prompt versions.
 * 
 * @property {number} id - The unique identifier of the prompt.
 * @property {string} name - The name of the prompt.
 * @property {string} createdAt - The timestamp when the prompt was created.
 * @property {string} updatedAt - The timestamp when the prompt was last modified.
 */
export interface Prompt {
    id: number;
    name: string;
    createdAt: string;
    updatedAt: string;
}